css-shared-element-transitions/index.bs

```diff
@@ -593,254 +655,307 @@ document's DOM structure.
-		The natural size of the image is equal to the |interest rectangle| bounds.
+    To <dfn lt="capture the image|capturing the image">capture the image</dfn> given an {{Element}} |element|, perform the following steps.
+    They return an image.
+
+    1. Render the referenced element and its descendants,
+        at the same size that they would be in the document,
+        over an infinite transparent canvas with the following characteristics:
+
+        * The origin of |element|'s [=ink overflow rectangle=] is anchored to canvas origin.
+
+        * If the referenced element has a transform applied to it (or its ancestors),
+            then the transform is ignored.
+
+            Note: This transform is applied to the snapshot using the `transform` property of the associated ''::page-transition-container'' pseudo-element.
+
+        * [=list/For each=] |descendant| of [=shadow-including descendant=] {{Element}} and [=pseudo-element=] of |element|,
+            if |descendant| has a [=computed value=] of 'page-transition-tag' that is not ''page-transition-tag/none'',
+            then skip painting |descendant|.
+
+            Note: This is necessary since the descendant will generate its own snapshot which will be displayed and animated independently.
+
+    1. Let |interestRectangle| be the result of [=computing the interest rectangle=] for |element|.
+
+        Note: The |interestRectangle| is the subset of |element|'s [=ink overflow rectangle=] that should be captured.
+            This is required for cases where an element's ink overflow rectangle needs to be clipped because of hardware constraints.
+            For example, if it exceeds the maximum texture size.
+
+    1. Return the portion of the canvas within |interestRectangle| as an image.
+        The natural size of the image is equal to the |interestRectangle| bounds.
 </div>
-	
+
 <div algorithm>
-	To <dfn>update transition DOM</dfn> given a {{SameDocumentTransition}} |manager|:
-	
-	1. For each [=page-transition pseudo-elements=] associated with |manager|,
-		check whether there is an active animation associated with this pseudo-element.
-	
-		Issue: Define what active animation means here.
-	
-	1. If no [=page-transition pseudo-elements=] has an active animation:
-	
-        1. Set |manager|&apos;s {{[[Phase]]}} internal slot to "idle".
-
-        1. [=Resolve=] |manager|&apos;s{{[[finished]]}} with |finishedP|.
+    To <dfn>update transition DOM</dfn> given a {{SameDocumentTransition}} |transition|:
+
+    1. For each [=page-transition pseudo-elements=] associated with |transition|,
+        check whether there is an active animation associated with this pseudo-element.
+
+        Issue: Define what active animation means here.
+
+    1. If no [=page-transition pseudo-elements=] has an active animation:
+
+        1. Set |transition|'s [=SameDocumentTransition/phase=] to "`finished`".
+
+        1. [=Resolve=] |transition|'s[=SameDocumentTransition/finished promise=].
```

Comment:
We also need to remove all pseudo-elements before resolving the finished promise.